pkg/config/autodiscover/autodiscover_targets.go

```diff
@@ -0,0 +1,142 @@
+// Copyright (C) 2021 Red Hat, Inc.
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+package autodiscover
+
+import (
+	log "github.com/sirupsen/logrus"
+	"github.com/test-network-function/test-network-function/pkg/config/configsections"
+	"github.com/test-network-function/test-network-function/pkg/tnf/testcases"
+)
+
+const (
+	configuredTestFile         = "testconfigure.yml"
+	operatorLabelName          = "operator"
+	skipConnectivityTestsLabel = "skip_connectivity_tests"
+)
+
+var (
+	operatorTestsAnnotationName    = buildAnnotationName("operator_tests")
+	subscriptionNameAnnotationName = buildAnnotationName("subscription_name")
+	podTestsAnnotationName         = buildAnnotationName("host_resource_tests")
+)
+
+// FindTestTarget builds a `configsections.TestTarget` from the current state of the cluster,
+// using labels and annotations to populate the data.
+func FindTestTarget(labels []configsections.Label) (target configsections.TestTarget) {
+	// find pods by label
+	for _, l := range labels {
+		pods, err := GetPodsByLabel(l)
+		if err == nil {
+			for i := range pods.Items {
+				target.PodsUnderTest = append(target.PodsUnderTest, buildPodUnderTest(&pods.Items[i]))
+				target.ContainersUnderTest = append(target.ContainersUnderTest, buildContainersFromPodResource(&pods.Items[i])...)
+			}
+		} else {
+			log.Warnf("failed to query by label: %v %v", l, err)
+		}
+	}
+	// Containers to exclude from connectivity tests are optional
+	identifiers, err := getContainerIdentifiersByLabel(configsections.Label{Namespace: tnfNamespace, Name: skipConnectivityTestsLabel, Value: anyLabelValue})
+	target.ExcludeContainersFromConnectivityTests = identifiers
+
+	if err != nil {
+		log.Warnf("an error (%s) occurred when getting the containers to exclude from connectivity tests. Attempting to continue", err)
+	}
+
+	csvs, err := GetCSVsByLabel(operatorLabelName, anyLabelValue)
+	if err == nil {
+		for i := range csvs.Items {
+			target.Operators = append(target.Operators, buildOperatorFromCSVResource(&csvs.Items[i]))
+		}
+	} else {
+		log.Warnf("an error (%s) occurred when looking for operaters by label", err)
+	}
+
+	return target
+}
+
+// buildPodUnderTest builds a single `configsections.Pod` from a PodResource
+func buildPodUnderTest(pr *PodResource) (cnf configsections.Pod) {
```

Comment:
all those functions with ``build`` and ``create`` names are little bit confusion.
When we should use one or the other ? Also, we're just fetching the data and populating the structure, we're not creating the container/pod itself.
not a comment, but wondering how we should address this

Reply:
I find them ok, given the context of the autodiscover package. And they are not exported. It would make the names too long if we add some suffix to every type, like podInfo or containerData.

Reply:
Also, I didn't find a function named createXyz in this package. If you are talking about the ones under test-network-function/common, they will be refactored as part of CNFCERT-81. We will make it more consistent then.

Reply:
Yes. I was talking in general. and ``buildXyzData`` is more appropriate I guess. ``CreateXyz`` gives the impression that we're creating a ``Pod`` or a ``container``.. but really we're only filling out structures and reading data from the ``deployment``

Reply:
Sure, those are all in test-network-function/common. Let's align them to a common naming convention once the refactoring is done by CNFCERT-81. I have a feeling that some of the functions may just get combined with others and there won't be as many names to worry about.